src/parser.rs

```diff
@@ -1147,16 +1248,48 @@ impl<'a> Parser<'a> {
         input
     }
 
+    fn last_slash_can_be_removed(serialization: &String, path_start: usize) -> bool {
+        let url_before_segment = &serialization[..serialization.len() - 1];
+        if let Some(segment_before_start) = url_before_segment.rfind("/") {
+            // Do not remove the root slash
+            segment_before_start >= path_start
+                // Or a windows drive letter slash
+                && !path_starts_with_windows_drive_letter(&serialization[segment_before_start..])
+        } else {
+            false
+        }
+    }
+
+    /// https://url.spec.whatwg.org/#shorten-a-urls-path
+    fn shorten_path(&mut self, scheme_type: SchemeType, path_start: usize) {
+        // If path is empty, then return.
+        if self.serialization.len() <= path_start {
```

Comment:
I don't understand how the comment relates to the condition.

Reply:
There are now three possibilities: 
- The scheme is special and the path is empty: "https://servo.org" ( path start is right after the buffer, self.serialization.len() )
- The scheme is not special and the path is empty: "tests://servo.org/" ( path start is the last slash position, self.serialization.len() -1 )
- The path is not empty: "https://servo.org/screenshot.png"  ( path start is the last slash position, which is much less than self.serialization.len() )

A URLs path is currently empty if ` self.serialization.len() < path_start `.
A URLs path only contains a root slash if ` self.serialization.len() == path_start `.

Reply:
If the comment makes sense to you here, it probably belongs to the code.
Let me know and I'll add it.
If not I need to figure out a more explicit way to convey that.

Reply:
So when is `path_start` ever greater than `self.serialization.len()`?

Reply:
I think you're correct, `if self.serialization.len() == path_start` is enough.